src/Framework/Constraint/Count.php

```diff
@@ -65,7 +66,7 @@ protected function getCountOf($other)
                 return $this->getCountOfGenerator($iterator);
             }
 
-            $key   = $iterator->key();
+            $key   = ($iterator instanceof Iterator) ? $iterator->key() : null;
```

Comment:
traversable = array or iterable follower, you need to ensure both are covered by tests